docs/index.md

```diff
@@ -35,6 +35,57 @@ non-public development, custom hardware handled by vendor
   source hardware and open instruction set architecture. Feel free to check the
   website about recent vPub plans or subscribe to newsletter.
 
+## Newcommers
+
+If you are new to Dasharo, this section it to help you get started using Dasharo
+firmware and possibly contributing to making it better.
+
+### Coreboot
+
+If you have no prior experience with coreboot, it's highly recommended to visit
+OpenSecurityTraining2 and finish these courses:
+* [Arch4031](https://p.ost2.fyi/courses/course-v1:OpenSecurityTraining2+Arch4031_x86-64_RV_coreboot+2021_v1/about),
+* [Arch4021](https://p.ost2.fyi/courses/course-v1:OpenSecurityTraining2+4021_Intro_UEFI+2022_v1/about)
+OpenSecurityTraining2 provides other great courses related to firmware, which
+are all freely avialable [here](https://p.ost2.fyi/courses).
+
+### Docker
+
+Coreboot needs a specific toolchain to be built, hence why it is usually built
+inside of a Docker. In case of Dasharo, all images are build using Docker, so
+you will make sure it works properly on your system.
+
+Follow these two links:
+* [Install Docker Engine on Ubuntu](https://docs.docker.com/engine/install/ubuntu/)
+* [Post-installation steps for Linux](https://docs.docker.com/engine/install/linux-postinstall/)
+
+### GitHub
+
+All code review and all issues related to Dasharo are resolved on
+[GitHub](https://github.com/). An account there is nessesary to contribute and
+report issues. All Dasharo repositories can be found
+[here](https://github.com/Dasharo).
+
+When you have an account on GitHub go ahead and
+[configure an SSH key](https://docs.github.com/en/authentication/connecting-to-github-with-ssh).
+It's also highly recommended to configure a GPG key before contributing
+antthing. Instructions to do that can be found
+[here](https://docs.github.com/en/authentication/managing-commit-signature-verification/adding-a-gpg-key-to-your-github-account).
+With that your account should be ready to contribute to Dasharo. Here are some
+additional links to help:
+* [Dasharo code structure](https://docs.dasharo.com/dev-proc/source-code-structure/)
+* [Creating a pull pequest from a fork](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork)
+* [Pull request eviews](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/about-pull-request-reviews)
```

Comment:
All above is valid, but little bit chaotic we essentially need tutorial which is step by step roadmap to first contribution to Dasharo, we also should link coreboot and UEFI contribution and explain how it is related to Dasharo. When enumerating steps we can link to precise components of OST2 courses to avoid duplication of information.

In short we prefer upstream first policy: https://www.chromium.org/chromium-os/chromiumos-design-docs/upstream-first/ - I guess we could write way better documentation for that in coreboot.
But also accepting PRs to our repos if for some reason (e.g. someone cannot sign [DCO](https://en.wikipedia.org/wiki/Developer_Certificate_of_Origin)) author cannot push upstream.

Reply:
I've changed this section a bit, when writing the assumption was that the reader has basic understanding of using git, but added commands to make sure the right branches and names are used.

But now this section makes me wonder: https://docs.dasharo.com/ways-you-can-help-us/#contribute-through-github, it's pretty vague. Maybe we should link this part of documentation there, so it's not just forking our repos?

Reply:
>  Maybe we should link this part of documentation there, so it's not just forking our repos?

Yes, that is a good idea

Reply:
@Stojek139808 please link and we can solve this conversation.

Reply:
Added the link there.